src/eos/radiationProperties/constant.cpp

```diff
@@ -1,22 +1,54 @@
 #include "constant.hpp"
 
-ablate::eos::radiationProperties::Constant::Constant(double absorptivity) : absorptivity(absorptivity) {}
+ablate::eos::radiationProperties::Constant::Constant(std::shared_ptr<eos::EOS> eosIn, double absorptivity, double emissivity)
+    : eos(std::move(eosIn)), absorptivityIn(absorptivity), emissivityIn(emissivity) {}
 
-PetscErrorCode ablate::eos::radiationProperties::Constant::ConstantFunction(const PetscReal *conserved, PetscReal *property, void *ctx) {
+PetscErrorCode ablate::eos::radiationProperties::Constant::ConstantAbsorptionFunction(const PetscReal conserved[], PetscReal *property, void *ctx) {
     PetscFunctionBeginUser;
     *property = *((double *)ctx);
     PetscFunctionReturn(0);
 }
-PetscErrorCode ablate::eos::radiationProperties::Constant::ConstantTemperatureFunction(const PetscReal *conserved, PetscReal temperature, PetscReal *property, void *ctx) {
+
+PetscErrorCode ablate::eos::radiationProperties::Constant::ConstantAbsorptionTemperatureFunction(const PetscReal conserved[], PetscReal temperature, PetscReal *property, void *ctx) {
     PetscFunctionBeginUser;
     *property = *((double *)ctx);
     PetscFunctionReturn(0);
 }
 
+PetscErrorCode ablate::eos::radiationProperties::Constant::ConstantEmissionFunction(const PetscReal conserved[], PetscReal *property, void *ctx) {
+    PetscFunctionBeginUser;
+    /** This model depends on mass fraction, temperature, and density in order to predict the absorption properties of the medium. */
+    auto functionContext = (FunctionContext *)ctx;
+    double temperature;  //!< Variables to hold information gathered from the fields
+    //!< Standard PETSc error code returned by PETSc functions
+
+    PetscCall(functionContext->temperatureFunction.function(conserved, &temperature, functionContext->temperatureFunction.context.get()));  //!< Get the temperature value at this location
+
+    PetscReal refractiveIndex = GetRefractiveIndex();  //! We may want to incorporate this at some point?
+    *(property) = functionContext->emissivity * ablate::radiation::Radiation::GetBlackBodyTotalIntensity(temperature, refractiveIndex);
+    PetscFunctionReturn(0);
+}
+
+PetscErrorCode ablate::eos::radiationProperties::Constant::ConstantEmissionTemperatureFunction(const PetscReal conserved[], PetscReal temperature, PetscReal *property, void *ctx) {
+    PetscFunctionBeginUser;
+    auto functionContext = (FunctionContext *)ctx;
+    PetscReal refractiveIndex = GetRefractiveIndex();  //! We may want to incorporate this at some point?
+    *(property) = functionContext->emissivity * ablate::radiation::Radiation::GetBlackBodyTotalIntensity(temperature, refractiveIndex);
```

Comment:
What is being returned from this function?  Why can't we compute the Emissivity and multiply by the total intensity elsewhere?